File: python/local-layer/handler.py
import time

from ddtrace import tracer

def handle(event, context):
    with tracer.trace('start'):
        time.sleep(0.01)
    with tracer.trace('importing'):
        import requests
    with tracer.trace('end'):
        time.sleep(0.01)
    return 'ok'

File: node/sns/handler.js
const tracer = require('dd-trace');

const AWS = require('aws-sdk');
const sns = new AWS.SNS();

const sleep = tracer.wrap('sleep', (ms) => {
  return new Promise(r => setTimeout(r, ms));
});

exports.producer = async function(event, context) {
  const params = {
    Message: JSON.stringify({ message: 'Hello from SNS' }),
    TopicArn: process.env.TOPIC_ARN
  };

  await sns.publish(params).promise();
  return params;
}

exports.consumer = async function(event, context) {
  console.log(JSON.stringify(event));
  await sleep(1000);
  return event;
}
